packages/bitcore-node/src/modules/firocoin/api/contract.ts

```diff
@@ -62,29 +65,71 @@ router.get('/:contractAddress', async (req, res) => {
 router.get('/:contractAddress/code', async (req, res) => {
   let { contractAddress } = req.params;
   try {
-    const fileExists = await fs.promises
-      .access(`${folderUpload}/${contractAddress}.sol`, fs.constants.F_OK)
-      .then(() => true)
-      .catch(() => false);
-    if (fileExists) {
-      res.download(`${folderUpload}/${contractAddress}.sol`);
-    } else {
-      res.status(404).send(`The requested contract address ${contractAddress} could not be found.`);
-    }
+    const data = await fetchGetContract(contractAddress);
+    await fs.promises.writeFile(`${folderUpload}/${contractAddress}.sol`, data.code, 'base64');
+    res.download(`${folderUpload}/${contractAddress}.sol`, `${contractAddress}.sol`, async (err) => {
+      if (err) {
+        throw err;
+      }
+      await fs.promises.unlink(`${folderUpload}/${contractAddress}.sol`);
+    });
   } catch (err) {
+    console.error(err);
     res.status(500).send(err);
   }
 });
 
 router.get('/:contractAddress/abi', async (req, res) => {
```

Comment:
Could we cache abi instead of re-compile? It would be more efficient in high load service

Reply:
yes krub